packages/cozy-mespapiers-lib/src/utils/search.js

```diff
@@ -0,0 +1,120 @@
+import { Document } from 'flexsearch'
+
+import { models } from 'cozy-client'
+
+import { CONTACTS_DOCTYPE } from '../doctypes'
+
+const {
+  document: {
+    themes: { themesList }
+  },
+  file: { isFile }
+} = models
+
+export const index = new Document({
+  document: {
+    id: '_id',
+    tag: 'flexsearchProps:tag',
+    index: [
+      'flexsearchProps:name',
+      'flexsearchProps:translatedQualificationLabel',
+      'flexsearchProps:fullname',
+      'flexsearchProps:birthday',
+      'flexsearchProps:birthcity',
+      'flexsearchProps:email',
+      'flexsearchProps:civility',
+      'flexsearchProps:company',
+      'flexsearchProps:jobTitle'
+    ],
+    store: ['_id', '_type']
+  }
+})
+
+export const makeFileTags = file => {
+  const item = file.metadata.qualification
+  const tags = themesList
+    .filter(theme => {
+      return theme.items.some(it => it.label === item.label)
+    })
+    .map(x => x.label)
+  return tags
+}
+
+export const makeContactTags = contact => {
+  const contactTags = []
+
+  const themesByAttributes = {
+    givenName: ['identity'],
+    familyName: ['identity'],
+    phone: ['home', 'work_study', 'identity'],
+    email: ['work_study', 'identity'],
+    cozy: ['identity'],
+    address: ['home', 'work_study', 'identity'],
+    birthday: ['identity'],
+    company: ['work_study'],
+    jobTitle: ['work_study']
+  }
+
+  Object.keys(themesByAttributes).map(attribute => {
+    const preAttribute = ['givenName', 'familyName'].includes(attribute)
+      ? 'name'
+      : undefined
+
+    const value = preAttribute
+      ? contact?.[preAttribute]?.[attribute]
+      : contact?.[attribute]
+
+    const themes = themesByAttributes[attribute]
+
+    if (value && value.length > 0) {
+      themes.forEach(theme => {
+        if (!contactTags.includes(theme)) {
+          contactTags.push(theme)
+        }
+      })
+    }
+  })
+
+  return contactTags
+}
+
+export const addFileDoc = (index, doc, scannerT) => {
+  return index.add({
+    ...doc,
```

Comment:
I don't think we want to index the *whole* doc, the resulting index will be huge, and we certainly don't need to search in every single attribute, especially for notes.

Reply:
on ne devrait donc indexer que les attributs sur lesquels on fait la recherche ?

Reply:
Oui, de la même façon qu'on indexe que les attributs strictement nécessaire avec une requête mango :wink: